spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRequiredDistributionAndOrdering.java

```diff
@@ -186,21 +186,22 @@ public void testDisabledDistributionAndOrdering() {
     Dataset<Row> inputDF = ds.coalesce(1).sortWithinPartitions("c1");
 
     // should fail if ordering is disabled
-    AssertHelpers.assertThrowsCause(
-        "Should reject writes without ordering",
-        IllegalStateException.class,
-        "Encountered records that belong to already closed files",
-        () -> {
-          try {
-            inputDF
-                .writeTo(tableName)
-                .option(SparkWriteOptions.USE_TABLE_DISTRIBUTION_AND_ORDERING, "false")
-                .option(SparkWriteOptions.FANOUT_ENABLED, "false")
-                .append();
-          } catch (NoSuchTableException e) {
-            throw new RuntimeException(e);
-          }
-        });
+    Assertions.assertThatThrownBy(
+            () -> {
+              try {
```

Comment:
the try-catch isn't needed and can be removed

Reply:
.append() is throwing NoSuchTableException

Reply:
this assertion check should be exactly like it's defined in https://github.com/apache/iceberg/blob/b7296a755ba9c3e9236a43f99225c580cd9abdc0/spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestRequiredDistributionAndOrdering.java#L165-L176. The fact that the code throws `NoSuchTableException` should not be an issue

Reply:
done. Thanks.